src/systemd/sd-id128.h

```diff
@@ -119,6 +120,37 @@ _sd_pure_ static __inline__ int sd_id128_is_allf(sd_id128_t a) {
 #define SD_ID128_NULL ((const sd_id128_t) { .qwords = { 0, 0 }})
 #define SD_ID128_ALLF ((const sd_id128_t) { .qwords = { UINT64_C(0xFFFFFFFFFFFFFFFF), UINT64_C(0xFFFFFFFFFFFFFFFF) }})
 
+_sd_pure_ static __inline__ int sd_id128_equalv(sd_id128_t a, va_list ap) {
+        int r = 0;
+
+        for (;;) {
+                sd_id128_t b = va_arg(ap, sd_id128_t);
+
+                if (sd_id128_is_null(b))
+                        break;
+
+                r = sd_id128_equal(a, b);
+                if (r)
+                        break;
+        }
+
+        return r;
+}
+
+_sd_pure_ static __inline__ int sd_id128_equals_sentinel(sd_id128_t a, ...) {
+        va_list ap;
+        int r;
+
+        va_start(ap, a);
+        r = sd_id128_equalv(a, ap);
+        va_end(ap);
+
+        return r;
+}
```

Comment:
uh, let's not do this. inline functions in public headers should remain the absolute exception, since they break encapsulation. I see no reason why this should be inline here.

Reply:
Please see the commit message. With an inline function the compiler is able to perform excellent optimization. With non-inline, it wouldn't be able to. I agree with the general principle, but in this case, by un-inlining we'd be making the code unoptimizable. (Without optimization, the code generated with this patch *is* quite a bit longer than status quo ante. With optimization on, it is becomes even shorter.) And OTOH, sd_id128_t is not an opaque structure, and what the function is completely trivial, so we're unlikely to be ever able to change behaviour. So overall, I think inline is justified here.

Reply:
does the optimization matter here though?

short code I like, good abstraction i like. micro-optimization though? hmpf...

Reply:
maybe move this to id128-util.h? i.e. my issue is with mixing inline + public. if you keep it internal inline is fine.